src/archutils/Win32/ExceptionHandler_x64.asm

```diff
@@ -0,0 +1,29 @@
+extrn ?MainExceptionHandler@@YAJPEAU_EXCEPTION_POINTERS@@@Z : PROC
```

Comment:
I would recommend against doing this. 

On x64 windows you'll have at least 2 pages (8k) of stack to handle a stack overflow which should be enough on it's own, but also allocating memory here could prevent the handling of OOM crashes. If you somehow actually need the extra space and can't/won't reduce the stack usage of your crash handlers, you should use a static array for extra memory instead VirtualAlloc.

Reply:
I was just trying to port the legacy code without further questioning its necessity. If you have got some better solution for both 32-bit and 64-bit cases, I will be happy to implement that.

Reply:
@sigatrev Please check my new commit to see if that is what it is supposed to be.

By the way, this adds 32 kilobytes to the binary size, but it should not be a problem with UPX compression.

Reply:
I definitely prefer this over the previous, but I still think you probably don't need to re-target the stack pointer at all.

Reply:
You mentioned before:

> On x64 windows you'll have at least 2 pages (8k) of stack to handle a stack overflow which should be enough on it's own

Then how much stack space will we have on 32-bit Windows?

Reply:
I believe its 4k on x86 windows

Another option that should work on both x86 and x64 is spinning up a new thread for the handler, which will have it's own stack.
```
    long ret = 0;
    std::thread thread([&]
    {
        ret = MainExceptionHandler(pExc);
    });
    thread.join();
    return ret;
```

Reply:
Nice. I will try your method and there will not be any ugly assembly code.